wrapper/src/main/java/software/amazon/jdbc/plugin/efm/MonitorThreadContainer.java

```diff
@@ -54,41 +54,35 @@ public static MonitorThreadContainer getInstance() {
   }
 
   static MonitorThreadContainer getInstance(final ExecutorServiceInitializer executorServiceInitializer) {
-    if (singleton == null) {
-      LOCK_OBJECT.lock();
-      try {
-        if (singleton == null) {
-          singleton = new MonitorThreadContainer(executorServiceInitializer);
-          CLASS_USAGE_COUNT.set(0);
-        }
-      } finally {
-        LOCK_OBJECT.unlock();
+    MonitorThreadContainer singletonToReturn;
+    LOCK_OBJECT.lock();
+    try {
+      if (singleton == null) {
+        singleton = new MonitorThreadContainer(executorServiceInitializer);
+        CLASS_USAGE_COUNT.set(0);
       }
+      singletonToReturn = singleton;
+      CLASS_USAGE_COUNT.getAndIncrement();
+    } finally {
+      LOCK_OBJECT.unlock();
     }
-    CLASS_USAGE_COUNT.getAndIncrement();
-    return singleton;
+    return singletonToReturn;
   }
 
   /**
    * Release resources held in the {@link MonitorThreadContainer} and clear references to the
    * container.
    */
   public static void releaseInstance() {
-    if (singleton == null) {
-      return;
-    }
-
-    if (CLASS_USAGE_COUNT.decrementAndGet() <= 0) {
-      LOCK_OBJECT.lock();
-      try {
-        if (singleton != null) {
-          singleton.releaseResources();
-          singleton = null;
-          CLASS_USAGE_COUNT.set(0);
-        }
-      } finally {
-        LOCK_OBJECT.unlock();
+    LOCK_OBJECT.lock();
```

Comment:
just double checking - do we enter the lock section even if singleton is null when method is called?

Reply:
Yes. I originally thought there could b a race condition in the current code where after the null check, and before it returns, another thread could call getInstance, which would make the singleton not null, and the original thread thats calling releaseInstance wouldnt know about that.
But now im wondering if thats an issue at all.

Reply:
I will add the null check and early return back. 
This isnt the area that we've seen issues with, so I think the safer approach here is to leave it as it was.